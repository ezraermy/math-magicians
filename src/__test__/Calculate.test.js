import calculate from '../logic/calculate';

describe('calculate', () => {
  it('returns an empty object for the "AC" button', () => {
    expect(calculate({ total: '10', next: '5', operation: '+' }, 'AC')).toMatchSnapshot();
  });

  it('updates the "next" value correctly for a number button', () => {
    expect(calculate({ next: '5', operation: '+' }, '6')).toMatchSnapshot();
  });

  it('updates the "next" value correctly for a "0" button', () => {
    expect(calculate({ next: '5', operation: '+' }, '0')).toMatchSnapshot();
  });

  it('updates the "next" value correctly for a "." button', () => {
    expect(calculate({ next: '5', operation: '+' }, '.')).toMatchSnapshot();
  });

  it('updates the "total" value correctly for an operation button', () => {
    expect(calculate({ total: '5', operation: '+' }, '-')).toMatchSnapshot();
  });

  it('updates the "operation" value correctly when pressing "="', () => {
    expect(calculate({ total: '5', next: '3', operation: '+' }, '=')).toMatchSnapshot();
  });

  it('updates the "next" value correctly for "+/-" button', () => {
    expect(calculate({ next: '5', operation: '+' }, '+/-')).toMatchSnapshot();
  });

  it('returns an empty object for an unknown button', () => {
    expect(calculate({ total: '5', next: '3', operation: '+' }, '#')).toMatchSnapshot();
  });
});
