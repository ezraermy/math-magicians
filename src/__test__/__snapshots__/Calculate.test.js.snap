// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate returns an empty object for an unknown button 1`] = `
Object {
  "next": null,
  "operation": "#",
  "total": "8",
}
`;

exports[`calculate returns an empty object for the "AC" button 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`calculate updates the "next" value correctly for "+/-" button 1`] = `
Object {
  "next": "-5",
  "operation": "+",
}
`;

exports[`calculate updates the "next" value correctly for a "." button 1`] = `
Object {
  "next": "5.",
  "operation": "+",
}
`;

exports[`calculate updates the "next" value correctly for a "0" button 1`] = `
Object {
  "next": "50",
  "operation": "+",
}
`;

exports[`calculate updates the "next" value correctly for a number button 1`] = `
Object {
  "next": "56",
  "operation": "+",
}
`;

exports[`calculate updates the "operation" value correctly when pressing "=" 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "8",
}
`;

exports[`calculate updates the "total" value correctly for an operation button 1`] = `
Object {
  "operation": "-",
  "total": "5",
}
`;
